lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/BuildJsonHttpDeserializer.kt

```diff
@@ -3,13 +3,14 @@ package com.swmansion.starknet.provider.rpc
 import com.swmansion.starknet.data.serializers.JsonRpcErrorSerializer
 import com.swmansion.starknet.provider.exceptions.RequestFailedException
 import com.swmansion.starknet.provider.exceptions.RpcRequestFailedException
+import com.swmansion.starknet.service.http.HttpResponse
 import com.swmansion.starknet.service.http.HttpResponseDeserializer
 import kotlinx.serialization.*
-import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.*
 import java.util.function.Function
 
 @Serializable
-private data class JsonRpcResponse<T>(
+internal data class JsonRpcResponse<T>(
     @SerialName("id")
     val id: Int,
```

Comment:
Since we're dealing with batch requests now, we should consider whether we ever wanna handle request with null `id`, as it is _technically_ possible per the [spec](https://www.jsonrpc.org/specification#response_object)

Reply:
In case of batch request, we shouldn't handle request with null `id` at all. Here's a quick example:
1. We send batch request 
2. We receive responses
3. Two or more of them have null `id`
4. We cannot identify them

Technically, if only one response has null `id`, we can identify it by elimination of others which are successful. However, this is a dirty solution and I don't think it's the way to go.

Wdyt @DelevoXDG ?

Reply:
Ok, I suppose since we're always sending `id` in our requests, we should always be getting `id` in the responses and this doesn't apply:

> If there was an error in detecting the id in the Request object (e.g. Parse error/Invalid Request), it MUST be Null.

Reply:
As we discussed offline, our requests will always include correct `id`, so we can assume that properly working server will also return responses with corresponding `id` values. In case response has `id`, the error is caused by the server and in such situation we don't want to handle the response.